### PR TITLE
Add a color eyedropper to the annotation editor

### PR DESCRIPTION
**Pull request by tusharmalpani20**
## Summary

  - Add an Eyedropper toolbar tool with the `I` shortcut.
  - Sample colors from the original full-resolution screenshot.
  - Correctly map cropped and high-DPI captures.
  - Keep right and bottom edge samples inside the selected crop.
  - Set the sampled color as the current custom annotation color.
  - Recolor a selected annotation with normal undo support.
  - Copy the uppercase `#RRGGBB` value to the clipboard.
  - Preserve the selected color if clipboard copying fails.
  - Return to Select after sampling.

  ## Testing

  - Added focused tests for cropped, high-DPI, right-edge, bottom-edge, and clamped sampling.
  - Added editor interaction and undo coverage.
  - Added focused smoke programs to Linux CI.
  - Full headless smoke suite passes.
  - `git diff --check` passes.

### CODE REVIEW

File: eyedropper.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      QRectF(QPointF(), previewSize));
+  if (boundedSelection.isEmpty())
+    return {};
+  const qreal u = std::clamp((displayPoint.x() - displayRect.left()) /
+                                 displayRect.width(), 0.0, 1.0);
+  const qreal v = std::clamp((displayPoint.y() - displayRect.top()) /
+                                 displayRect.height(), 0.0, 1.0);
+  const auto sourcePixel = [](qreal start, qreal end, qreal position,
+                              int sourceExtent, qreal previewExtent) {
+    const int first = std::clamp(
```
Comment (tobi) [inline eyedropper.cpp:23]: **[P2] Preserve fractional crop geometry when mapping the cursor**

Rounding both selection edges and then distributing `u` across that widened integer span changes the coordinate transform for fractional crops. For example, with a 2x source scale, selection `[0.1, 16.1)` maps to source `[0.2, 32.2)`. At `u = 0.99`, the displayed/source coordinate is `floor(0.2 + 0.99 * 32) = 31`, but this code uses `first=0`, `pastLast=33` and returns pixel 32. Crop handles routinely produce fractional `QRectF` edges, so clicks near an edge can copy the adjacent pixel. Compute the continuous source coordinate from `start + position * (end - start)`, scale it, then floor/clamp; please add a fractional-boundary case to the focused smoke test.

File: editor.cpp [hunk 2]
```diff
@@ -1370,6 +1379,29 @@ void CaptureEditor::mousePressEvent(QMouseEvent *event) {
   if (!editImageRect().contains(cursor_))
     return;
 
+  if (tool_ == Tool::Eyedropper) {
+    customColor_ = sampleSourceColor(capture_.source, capture_.preview.size(),
```
Comment (tobi) [inline editor.cpp:1383]: **[P2] Use an opaque RGB annotation color**

`pixelColor()` preserves alpha, and this `QColor` becomes the current annotation color. In file-edit mode, sampling a transparent or semi-transparent PNG therefore makes subsequent annotations transparent or unexpectedly faded even though the clipboard contract is `#RRGGBB`. Normalize the sampled color to alpha 255 before storing/applying it, and cover an ARGB source pixel in the focused test.

**Review by tobi — CHANGES_REQUESTED** [review]
Expanded validation passed in Crabbox on Cloudflare `standard-4`: Debug build with GCC 16.2.1, ASan+UBSan, and every `omasnap-*-smoke` executable completed with exit 0 (39.916s command, 41.680s total). Requesting changes for two uncovered color-correctness cases: fractional crop boundaries and source alpha.

**Comment by tobi** [comment]
Not merging. Two correctness issues on normal file-edit/crop paths:

- rounding fractional crop edges can sample the adjacent source pixel
- preserving source alpha can make annotations faded or invisible on transparent PNGs

Fix those (with regression cases) before this can land.

**Comment by tobi** [comment]
Landed eyedropper. Samples stay inside the crop and drop source alpha so annotations stay opaque.

**Comment by tobi** [comment]
Closing: this is already on main (applied directly after the branch diverged).
